### PR TITLE
feat: rename sidebar to sidebar-nav, use fc and switch to slots instead of props

### PR DESCRIPTION
**Pull request by LukasHirt**
**Change: Rename OcSidebar to OcSidebarNav**

We've changed the name of the `OcSidebar` component to `OcSidebarNav` to better reflect that it shouldn't be used for anything else than the navigation.

**Change: Use slots in OcSidebarNav**

We've moved away from defining the nav items in the `OcSidebarNav` via props to including them though slots.
The component now contains 3 slots - `header`, `nav` and `footer`.

**Enhancement: Add inverse button variant**

We've added inverse button variant which should be used on a dark background.

### CODE REVIEW

File: src/components/OcSidebarNav.vue [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+}
+</style>
+
+<docs>
+Sidebar navigation component containing branding
+
+## Examples
+```vue
+<oc-sidebar class="uk-height-large">
```
Comment (kulmann) [inline src/components/OcSidebarNav.vue:79]:
```suggestion
<oc-sidebar-nav class="uk-height-large">
```

File: src/components/OcSidebarNav.vue [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    <oc-list>
+      <oc-sidebar-nav-item icon="folder">Files</oc-sidebar-nav-item>
+      <oc-sidebar-nav-item :active="true" icon="shared-with-me">Shared with me</oc-sidebar-nav-item>
+      <oc-sidebar-nav-item icon="delete">Deleted files</oc-sidebar-nav-item>
+    </oc-list>
+  </template>
+  <template #footer>
+    <p class="oc-px-s">Made by ownClouders</p>
+  </template>
+</oc-sidebar>
```
Comment (kulmann) [inline src/components/OcSidebarNav.vue:97]:
```suggestion
</oc-sidebar-nav>
```

**Review by kulmann — CHANGES_REQUESTED** [review]

**Comment by LukasHirt** [comment]
@kulmann I applied your suggestions. Thanks for those catches!

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**